File: cpp/divisors.cpp
vector<long long> divisors(long long n){
    vector<long long> ans;
    for(long long i = 1;i*i <= n; ++i){
        if(n%i)continue;
        if(i*i == n){
            ans.push_back(i);
        }else{
            ans.push_back(i);
            ans.push_back(n/i);
        }
    }

    return ans;
}

File: rust/divisors.rs
fn divisors(n: i64) -> Vec<i64> {
    let mut retu = vec![];
    for i in 1.. {
        if i*i > n { break; }
        if n%i != 0 { continue; }
        if i*i == n {
            retu.push(i);
        } else {
            retu.push(i);
            retu.push(n/i);
        }
    }

    retu
}
